tools/serverpod_cli/lib/src/util/model_helper.dart

```diff
@@ -22,46 +30,93 @@ class ModelHelper {
   static Future<List<ModelSource>> loadProjectYamlModelsFromDisk(
     GeneratorConfig config,
   ) async {
-    List<FileSystemEntity> protocolSourceFileList = [];
-    try {
-      var protocolSourceDir =
-          Directory(joinAll(config.protocolSourcePathParts));
-      protocolSourceFileList =
-          await protocolSourceDir.list(recursive: true).toList();
-    } on PathNotFoundException catch (_) {}
+    var modelSources = <ModelSource>[];
 
-    List<FileSystemEntity> modelSourceFileList = [];
-    try {
-      var modelSourceDir = Directory(joinAll(config.modelSourcePathParts));
-      modelSourceFileList = await modelSourceDir.list(recursive: true).toList();
-    } on PathNotFoundException catch (_) {}
+    var modelSource = await _loadYamlModelsFromDisk(
+        defaultModuleAlias, _absolutePathParts(config.modelSourcePathParts));
+    modelSources.addAll(modelSource);
 
-    var sourceFileList = [...protocolSourceFileList, ...modelSourceFileList];
+    modelSource = await _loadYamlModelsFromDisk(
+        defaultModuleAlias, _absolutePathParts(config.protocolSourcePathParts));
+    modelSources.addAll(modelSource);
+
+    for (var module in config.modulesDependent) {
+      modelSource = await _loadYamlModelsFromDisk(
+          module.nickname, module.modelSourcePathParts);
+      modelSources.addAll(modelSource);
+
+      modelSource = await _loadYamlModelsFromDisk(
+          module.nickname, module.protocolSourcePathParts);
+      modelSources.addAll(modelSource);
+    }
 
     // TODO This sort is needed to make sure all generated methods
     // are in the same order. Move this logic to the code generator instead.
-    sourceFileList.sort((a, b) => a.path.compareTo(b.path));
+    modelSources
+        .sort((a, b) => a.yamlSourceUri.path.compareTo(b.yamlSourceUri.path));
 
-    var files = sourceFileList.whereType<File>().where(
-        (file) => modelFileExtensions.any((ext) => file.path.endsWith(ext)));
+    return modelSources;
+  }
+
+  static List<String> _absolutePathParts(List<String> pathParts) {
+    return split(absolute(joinAll(pathParts)));
+  }
+
+  static Future<List<ModelSource>> _loadYamlModelsFromDisk(
+    String moduleAlias,
+    List<String> pathParts,
+  ) async {
+    var files = await _loadAllModelFiles(pathParts);
 
     List<ModelSource> sources = [];
     for (var model in files) {
       var yaml = await model.readAsString();
 
       sources.add(ModelSource(
+        moduleAlias,
         yaml,
         model.uri,
-        extractPathFromModelRoot(config, model.uri),
+        extractPathFromModelRoot(pathParts, model.uri),
       ));
     }
 
     return sources;
   }
 
+  static Future<Iterable<File>> _loadAllModelFiles(
+    List<String> absolutePathParts,
+  ) async {
+    List<FileSystemEntity> modelSourceFileList = [];
+
+    var path = joinAll(absolutePathParts);
+
+    if (!isAbsolute(path)) {
+      path =
+          Platform.isWindows ? '$separator$separator$path' : '$separator$path';
```

Comment:
Question: This really does look sketchy. Are you sure there is no path or directory library support to start a path from the root directory? Seems like a fairly common scenario that should be supported on all plattforms.

Reply:
there is `absolute()` but it converts the path inject to the current directory. I have not found any other method, I have taken the convention from the `isAbsolute` documentation comment that an absolute path starts with / on POSX system and \\ on windows.

Reply:
OK! Still feels strange that there is no built in support for creating this.